api/src/lib/crud.js

```diff
@@ -0,0 +1,128 @@
+import koa from 'koa';
+import coBody from 'co-body';
+import koaRoute from 'koa-route';
+
+export default (queries, restricted, excludedRouteMethod) => {
+    const app = koa();
+    let q;
+
+    app.use(function* (next) {
+        q = queries(this.client);
+        yield next;
+    });
+
+    if (!excludedRouteMethod || excludedRouteMethod.indexOf('get') === -1) {
```

Comment:
I cleaned this file and append this middleware
```js
app.use(function* (next) {
    if (availableMethods.indexOf(this.method) === -1) {
        this.status = 405; // HTTP 405 Method Not Allowed
        this.body = {};
    }

    yield next;
});
```

Let me know if it suits you.